dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt

```diff
@@ -81,14 +83,64 @@ interface CompanyApi {
     @PreAuthorize("hasRole('ROLE_USER')")
     fun getCompanies(
         @RequestParam searchString: String? = null,
-        @RequestParam dataTypes: Set<DataType>? = null,
+        @RequestParam(required = true) dataTypes: Set<DataType>,
         @RequestParam countryCodes: Set<String>? = null,
         @RequestParam sectors: Set<String>? = null,
         @RequestParam onlyCompanyNames: Boolean = false,
         @RequestParam onlyCurrentUserAsUploader: Boolean = false,
+
     ):
         ResponseEntity<List<StoredCompany>>
 
+    /**
+     * A method to retrieve companies with names or identifiers matching a search string
+     * @param searchString string used for substring matching in the name and the identifiers of a company
+     * @return names of the first 100 companies matching the search criteria
+     */
+    @Operation(
+        summary = "Retrieve specific companies by searching their names and identifiers",
+        description = "Companies identified via the provided company name/identifier are retrieved",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved company names."),
+        ],
+    )
+    @GetMapping(
+        value = ["/names"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    fun getCompaniesBySearchString(
+        @RequestParam searchString: String,
+    ):
+        ResponseEntity<List<CompanyIdAndName>>
+
+    /**
+     * A method to check if an identifier of a given type exists
+     * @param identifierType the type of the identifier
+     * @param identifier the identifier
+     */
+    @Operation(
+        summary = "Checks that an identifier of specified type exists.",
+        description = "Checks that an identifier of specified type exists.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully checked that identifier exists."),
+            ApiResponse(responseCode = "404", description = "Successfully checked that identifier does not exist."),
+        ],
+    )
+    @RequestMapping(
+        method = [org.springframework.web.bind.annotation.RequestMethod.HEAD],
```

Comment:
In the `DocumentApi.kt`, the method is described shorter with just 
`method = RequestMethod.HEAD`<

Maybe here too?

Reply:
FYI:

We could also use the `@HeadMapping` annotation instead of setting the method in the `@RequestMapping` annotiaton.
That would be more consistent to our other endpoint which are not HEAD endpoints.

Smth like:
`@HeadMapping("/identifiers/{identifierType}/{identifier}")`

Reply:
1. Ist mir nicht aufgefallen, hab es verkürzt.
2. @HeadMapping findet meine IDE nicht.

Reply:
1. Sicher dass der change gepusht wurde? Sehe ihn nicht.

2. Kann sein dass das deprecated ist. Sorry.  Dann passt's wie es jetzt ist.

Reply:
👍